imap/message.c

```diff
@@ -985,6 +986,74 @@ static int read_headers_condstore_qresync_updates(struct ImapAccountData *adata,
 
   return 0;
 }
+
+/**
+ * imap_verify_qresync - Check to see if QRESYNC got jumbled
+ * @param m  Imap Selected Mailbox
+ * @retval  0 Success
+ * @retval -1 Error
+ *
+ * If so, wipe the context and try again with a normal download.
+ */
+static int imap_verify_qresync(struct Mailbox *m)
+{
+  assert(m);
+  struct ImapAccountData *adata = imap_adata_get(m);
+  struct ImapMboxData *mdata = imap_mdata_get(m);
+  if (!adata || (adata->mailbox != m))
+    return -1;
+
+  const size_t max_msn = imap_msn_highest(&mdata->msn);
+
+  unsigned int msn;
+  unsigned int uid;
+  struct Email *e = NULL;
+  struct Email *uidh = NULL;
+
```

Comment:
I've moved the `int i;` definition into the `for` (and below too), to reduce `i`'s scope.